lib/sober_swag/serializer/array.rb

```diff
@@ -1,30 +1,53 @@
 module SoberSwag
   module Serializer
     ##
-    # Make a serialize of arrays out of a serializer of the elements
+    # Transform a serializer that works on elements to one that works on Arrays
     class Array < Base
+      ##
+      # Make an array serializer out of another serializer.
+      # @param element_serializer [SoberSwag::Serializer::Base] the serializer to use for each element.
       def initialize(element_serializer)
         @element_serializer = element_serializer
       end
 
+      ##
+      # The serializer that will be used for each element in the array.
+      #
+      # @return [SoberSwag::Serializer::Base]
+      attr_reader :element_serializer
+
+      ##
+      # Delegates to {#element_serializer}
       def lazy_type?
         @element_serializer.lazy_type?
       end
 
+      ##
+      # Delegates to {#element_serializer}, wrapped in an array
       def lazy_type
         SoberSwag::Types::Array.of(@element_serializer.lazy_type)
       end
 
+      ##
+      # Delegates to {#element_serializer}
       def finalize_lazy_type!
         @element_serializer.finalize_lazy_type!
       end
 
-      attr_reader :element_serializer
-
+      ##
+      # Serialize an array of objects that can be serialized with {#element_serializer}
+      # by calling `element_serializer.serialize` for each item in this array.
+      #
+      # Note: since ruby is duck-typed, anything that responds to {#map} should work!
```

Comment:
Do all comments that mention the following, refer to [SoberSwag::Nodes::Base#map](https://github.com/SonderMindOrg/sober_swag/blob/30f39844650ea96eacab4417ee43884cc799be48/lib/sober_swag/nodes/base.rb#L64)?
> anything that responds to {#map} should work!

Reply:
No! I'll clarify.